doc/user_requirements.md

```diff
@@ -0,0 +1,236 @@
+# System Requirement Specification (SRS) Virtual Schema Common Java (VSCJ) 
+
+This document contains the system requirements for the Java bottom-layer base library for all Exasol Virtual Schema adapters.
+
+## Features
+
+### Property Validation
+`feat~property-validation~1`
+
+The VSCJ library provides the infrastructure for validating the input coming from virtual schema properties.
+
+Rationale:
+
+The validation on the one hand provides users with better information about properties that have been set incorrectly and on the other hand improves the security by preventing faulty or malicious property values to endanger the VS operation.
+
+Needs: req
+
+## High-level Requirements
+
+### Property Validation
+
+Users provide properties with virtual schema definitions that serve as configuration.
+
+Here is a non-exhaustive list of typical properties that should give an idea of what they are used for:
+
+1. Name under which the virtual schema can be found in the Exasol database
+2. ID of the connection object that stores the access data to the data source
+3. Log level
+4. Target host for logs
+5. Connection type selector
+6. Feature switch
+
+Please refer to [Appendix A — Known Virtual Schema Property Types](#appendix-a--known-virtual-schema-property-types) for a more complete list.
+
+Note that some of these properties have relationships that require to validate them together. In the Exasol Virtual Schema for instance, the selected connection type decides which other connection properties are required.
+
+Also, some of the validations are so dialect-specific, that covering them in the VSCJ base library is not reasonable.
+
+#### Validating the Existence of Mandatory Properties
+`req~validating-the-existence-of-mandatory-properties~1`
+
+VSCJ allows validating that a mandatory virtual schema property is set.
```

Comment:
This sentence is a bit confusing. Maybe rephrase it.

Reply:
Maybe:
```
VSCJ must implement a validation that a mandatory virtual schema property is set.
```

Reply:
Fixed.